src/launcher/features/filter/UpdateAllApps.tsx

```diff
@@ -7,17 +7,17 @@
 import React from 'react';
 import Button from 'react-bootstrap/Button';
 
-import { upgradeDownloadableApp } from '../../actions/appsActions';
-import { getUpgradeableVisibleApps } from '../../reducers/appsReducer';
 import { useLauncherDispatch, useLauncherSelector } from '../../util/hooks';
+import { upgradeDownloadableApp } from '../apps/appsEffects';
+import { getUpgradeableVisibleApps } from '../apps/appsSlice';
 
 export default () => {
     const dispatch = useLauncherDispatch();
     const upgradeableApps = useLauncherSelector(getUpgradeableVisibleApps);
 
     const upgradeAllApps = () =>
-        upgradeableApps.forEach(({ name, latestVersion, source }) =>
-            dispatch(upgradeDownloadableApp(name, latestVersion, source))
+        upgradeableApps.forEach(app =>
+            dispatch(upgradeDownloadableApp(app, app.latestVersion))
```

Comment:
Could this api change it's second parameter to match the app.latestVersion as a default value? Seems like a lot of cases just want to do something like `upgradeDownloadableApp(app)`

Reply:
Without investigating it: I cannot think of any case where we currently upgrade to a specific version instead of “the latest one”. So, yes, this probably can be simplified. :-) I came across lots of small things like this but didn't want to change too many of them. Otherwise it would have gotten even bigger and taken even longer.